RetailCoder.VBE/Inspections/QuickFixes/AssignedByValParameterMakeLocalCopyQuickFix.cs

```diff
@@ -97,13 +97,42 @@ private void ReplaceAssignedByValParameterReferences(IModuleRewriter rewriter, s
             }
         }
 
+
         private void InsertLocalVariableDeclarationAndAssignment(IModuleRewriter rewriter, string localIdentifier)
         {
-            var content = Tokens.Dim + " " + localIdentifier + " " + Tokens.As + " " + _target.AsTypeName + Environment.NewLine
-                + (_target.AsTypeDeclaration is ClassModuleDeclaration ? Tokens.Set + " " : string.Empty)
-                + localIdentifier + " = " + _target.IdentifierName;
+            var content = Tokens.Dim + " " + localIdentifier + " " + Tokens.As + " " + _target.AsTypeName + Environment.NewLine;
+            if (IsBaseTypeContext(_target))
+            {
+                content = content + localIdentifier + " = " + _target.IdentifierName;
+            }
+            else 
+            {
+                //All we can know is that it is not a Base type.  Let VBA determine
+                //the right way to assign the parameter.  The user can simplify it later.
+                string insertIsObjectCheck =
+@"If(IsObject({1})) Then
```

Comment:
@BZngr it's worrying that an enum triggers the error - that's definitely a bug, since the inspection is explicitly handling `Enum` and `Type` types. Looks like the Boolean logic for that inspection needs to be clarified, it's a bit of a mess ATM (probably my fault...).